reportportal_client/utilities.py

```diff
@@ -0,0 +1,137 @@
+import six
+from time import time
+
+from .errors import ResponseError, EntryCreatedError, OperationCompletionError
+
+
+def _convert_string(value):
+    """Support and convert strings in py2 and py3.
+
+    :param value: input string
+    :return value: convert string
+    """
+    if isinstance(value, six.text_type):
+        # Don't try to encode 'unicode' in Python 2.
+        return value
+    return str(value)
+
+
+def _dict_to_payload(dictionary):
+    """Convert dict to list of dicts.
+
+    :param dictionary: initial dict
+    :return list: list of dicts
+    """
+    system = dictionary.pop("system", False)
+    return [
+        {"key": key, "value": _convert_string(value), "system": system}
+        for key, value in sorted(dictionary.items())
+    ]
+
+
+def _get_id(response):
+    """Get id from Response.
+
+    :param response: Response object
+    :return id: int value of id
+    """
+    try:
+        return _get_data(response)["id"]
+    except KeyError:
+        raise EntryCreatedError(
+            "No 'id' in response: {0}".format(response.text))
+
+
+def _get_msg(response):
+    """
+    Get message from Response.
+
+    :param response: Response object
+    :return: data: json data
+    """
+    try:
+        return _get_data(response)
+    except KeyError:
```

Comment:
Where would this error come from? I can't see where that might get raised.

Reply:
This is dealt with by extracting duplicated code to create new method.